internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -645,6 +729,30 @@ func flattenMonitorDiagnosticEnabledLogs(input *[]diagnosticsettings.LogSettings
 	return enabledLogs
 }
 
+func flattenMonitorDiagnosticEnabledMetrics(input *[]diagnosticsettings.MetricSettings) []interface{} {
+	enabledLogs := make([]interface{}, 0)
+	if input == nil {
+		return enabledLogs
+	}
+
+	for _, v := range *input {
+		output := make(map[string]interface{})
+
+		if !v.Enabled {
+			continue
+		}
+
+		category := ""
```

Comment:
This block can be simplified to
```suggestion
		output["category"] = pointer.From(v.Category)
```